File: cakeymccakeface.py
from sys import stdin,stdout

N,M = int(stdin.readline()),int(stdin.readline())
entries,exits = list(map(int,stdin.readline().split())),list(map(int,stdin.readline().split()))
time_difference = {}
#O(N*M)
for a in entries:
    for b in exits:
        if b - a >= 0:
            time_difference[b-a] = 1 if b - a not in time_difference else time_difference[b-a] + 1
secret,highest_value = 0,0
for key,value in time_difference.items():
    if value > highest_value:
        secret = key
        highest_value = value
    if value == highest_value and key < secret:
        secret = key
stdout.write(f"{secret}")

File: cakeymccakeface.java
import java.util.*;
import java.io.*;

public class cakeymccakeface {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        int N = Integer.parseInt(br.readLine()), M = Integer.parseInt(br.readLine());
        int[] entries = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray(); 
        int[] exits = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray(); 
        HashMap<Integer,Integer> timeDifference = new HashMap<>();
        //O(N*M)
        for (int a: entries){
            for (int b: exits){
                if (b - a >= 0){
                    if (timeDifference.containsKey(b-a)) timeDifference.put(b - a, timeDifference.get(b-a) + 1);
                    else timeDifference.put(b - a, 1);
                }
            }
        }
        int answer = 0, highestFrequency = 0;
        for (Map.Entry<Integer,Integer> entry: timeDifference.entrySet()){
            if (entry.getValue() > highestFrequency){
                answer = entry.getKey();
                highestFrequency = entry.getValue();
            }
            if (entry.getValue() == highestFrequency && entry.getKey() < answer){
                answer = entry.getKey();
            }
        }
        pw.println(answer);
        br.close();
        pw.close();
    }
}
